sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -118,12 +119,19 @@ case class BroadcastExchangeExec(
           // SparkFatalException, which is a subclass of Exception. ThreadUtils.awaitResult
           // will catch this exception and re-throw the wrapped fatal throwable.
           case oe: OutOfMemoryError =>
-            throw new SparkFatalException(
+            val sizeMessage = if (dataSize != -1) {
+              s"; Size of table is $dataSize"
```

Comment:
You are taking into account only size of the `relation`.  Probably, result of `child.executeCollectIterator()` consumes some memory too, and an user will need more memory than `dataSize`?

Reply:
I think you're right @MaxGekk.  I was thinking there was a MemoryManager involved here, and you could query its size, but now I see thats only in `HashedRelation.apply()`.

Nonetheless -- this is still useful info to include in the message, right?  It would be better if we accurately tracked the total size needed, but this is still helpful.

Reply:
> this is still useful info to include in the message, right?

No doubt, I would slightly improve the error message and mention that driver memory should be increased at least by `dataSize` or maybe more. And `Size of table ...` -> `Estimated size of hashed relation ...to be broadcasted`

Reply:
+1